app/layout.tsx

```diff
@@ -23,10 +24,12 @@ export default function RootLayout({
   return (
     <html lang='en'>
```

Comment:
```suggestion
// import cookies from next/headers in the layout and grab the theme and assign it to the html of the layout
  const cookieStore = cookies();
  const theme = cookieStore.get('theme');
  // ...
  
    <html lang='en' className={theme?.value}>
```

This is why we use cookies for this. So that the server can access the theme state before rendering anything.